### PR TITLE
initial AsyncIO implementation

### PR DESCRIPTION
**Pull request by clinuxrulz**
https://github.com/highj/highj/issues/61

### CODE REVIEW

File: src/main/java/org/highj/data/stateful/AsyncIO.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import org.derive4j.hkt.__;
+import org.derive4j.hkt.__2;
+import org.highj.data.Either;
+import org.highj.data.Maybe;
+import org.highj.data.stateful.async_io.*;
+import org.highj.data.tuple.T0;
+import org.highj.function.F1;
+
+public interface AsyncIO<E,A> extends __2<AsyncIO.µ,E,A> {
+    enum µ {}
```
Comment (DanielGronau) [inline src/main/java/org/highj/data/stateful/AsyncIO.java]: Does `enum` has an advantage over `interface` here, other than having 5 chars less :)

File: src/main/java/org/highj/data/stateful/AsyncIO.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+import org.derive4j.hkt.__;
+import org.derive4j.hkt.__2;
+import org.highj.data.Either;
+import org.highj.data.Maybe;
+import org.highj.data.stateful.async_io.*;
+import org.highj.data.tuple.T0;
+import org.highj.function.F1;
+
+public interface AsyncIO<E,A> extends __2<AsyncIO.µ,E,A> {
```
Comment (DanielGronau) [inline src/main/java/org/highj/data/stateful/AsyncIO.java]: Maybe you can add a short javadoc description how `AsyncIO` should be used, and what the restrictions are.

File: src/main/java/org/highj/data/stateful/AsyncIO.java (hunk 1 above)
Comment (clinuxrulz) [inline src/main/java/org/highj/data/stateful/AsyncIO.java]: no advantage at all. We can change it to ```interface``` for a bit of consistency.

**Comment by clinuxrulz** [comment]
a ```fork``` method and the ability to cancel a running async computation would be nice too, down the track. I'd follow purescripts ```Aff``` monad as a guide.

**Pull request merged**